modules/http4s/src/main/scala/natchez/http4s/syntax/EntryPointOps.scala

```diff
@@ -15,21 +15,70 @@ import cats.effect.Resource
 import natchez.TraceValue
 import cats.Monad
 import org.http4s.server.websocket.WebSocketBuilder2
-
+import org.typelevel.ci.CIString
+
+/**
+ * @define liftT
+ *         Given an entry point and HTTP Routes in Kleisli[F, Span[F], *] return routes in F. A new span
+ *         is created with the URI path as the name, either as a continuation of the incoming trace, if
+ *         any, or as a new root.
+ *
+ * @define liftR
+ *         Lift an `HttpRoutes`-yielding resource that consumes `Span`s into the bare effect. We do this
+ *         by ignoring any tracing that happens during allocation and freeing of the `HttpRoutes`
+ *         resource. The reasoning is that such a resource typically lives for the lifetime of the
+ *         application and it's of little use to keep a span open that long.
+ *
+ * @define wsLiftT
+ *         Given an entry point and a function from `WebSocketBuilder2` to HTTP Routes in
+ *         Kleisli[F, Span[F], *] return a function from `WebSocketBuilder2` to routes in F. A new span
+ *         is created with the URI path as the name, either as a continuation of the incoming trace, if
+ *         any, or as a new root.
+ *
+ * @define wsLiftR
+ *         Lift a `WebSocketBuilder2 => HttpRoutes`-yielding resource that consumes `Span`s into the bare
+ *         effect. We do this by ignoring any tracing that happens during allocation and freeing of the
+ *         `HttpRoutes` resource. The reasoning is that such a resource typically lives for the lifetime
+ *         of the application and it's of little use to keep a span open that long.
+ *
+ * @define headersInfo
+ *         All headers except security (Authorization, Cookie, Set-Cookie)
+ *         and payload (Content-Length, ContentType, Content-Range, Trailer, Transfer-Encoding)
+ *         are passed to Kernel.
+ *
+ * @define isKernelHeader should an HTTP header be passed to Kernel or not
+ */
 trait EntryPointOps[F[_]] { outer =>
 
   def self: EntryPoint[F]
 
   /**
-   * Given an entry point and HTTP Routes in Kleisli[F, Span[F], *] return routes in F. A new span
-   * is created with the URI path as the name, either as a continuation of the incoming trace, if
-   * any, or as a new root.
+   * $liftT
+   *
+   * @note $headersInfo
    */
   def liftT(routes: HttpRoutes[Kleisli[F, Span[F], *]])(
     implicit ev: MonadCancel[F, Throwable]
   ): HttpRoutes[F] =
+    liftT(routes, name => !EntryPointOps.ExcludedHeaders.contains(name))
+
+  /**
+   * $liftT
+   *
+   * @param isKernelHeader $isKernelHeader
+   */
+  def liftT(
+    routes: HttpRoutes[Kleisli[F, Span[F], *]],
+    isKernelHeader: CIString => Boolean
```

Comment:
Is there a reason this couldn't be a default argument, rather than adding overloads? I don't really care, was just curious if you tried it.

Reply:
Default argument works as well. 

Initially, I was thinking about a new trait `EntryPointOpsBinCompat` (or something similar) to prevent binary-breaking changes.

AFAIK default arguments are binary-breaking. If there is no plan for a deprecation cycle I can replace overloads with default args.

Reply:
I'll just bump the version to 0.3.0 … I'm not really sweating bincompat yet.

Reply:
Sounds good. Then I will use default arguments. I will push an updated version soon.